api/nakadi-event-bus-api.yaml

```diff
@@ -0,0 +1,1118 @@
+            top level of the Event. If name conflicts arise, creation of this EventType will be rejected.
+            Default validation_strategy for this `EventType` is `[{name: 'schema-validation'}]`.
+      effective_schema:
+        type: string
+        #read-only: true
+        description: |
+          The effective schema of this `EventType`. The predefined schema validator will use the value of
+          this field as base for the validation.
+
+          The creation of EventTypes of different categories implies a distinct general structure for the
+          payload of events of each type. This ultimate format the payload must adhere to is reflected in
+          the `effective_schema`. See description of field `category` for more details of its generation.
+
+          This field is informative and its contents cannot be manipulated directly.
+
+      validation_strategies:
+        description: |
+          Determines the validation that has to be executed upon reception of Events of this type. Events are
+          rejected if any of the rules fail (see details of Problem response on the Event publishing
+          methods).
+
+          Rule evaluation order is the same as in this array.
+
+          If not explicitly set, default value will respect the definition of the `EventType.category`.
+        type: array
+        items:
+          $ref: '#/definitions/EventValidationStrategy'
+
+      enrichment_strategies:
+        description: |
+          Determines the enrichment to be performed on an Event upon reception. Enrichment is performed once
+          upon reception (and after validation) of an Event and is only possible on fields that are not
+          defined on the incoming Event.
+
+          See documentation for the write operation for details on behaviour in case of unsuccessful
+          enrichment.
+        type: array
+        items:
+          $ref: '#/definitions/EventEnrichmentStrategy'
+        default: [{name: 'metadata-enrichment'}]
+
+      partition_resolution_strategy:
+        description: |
+          Determines how the assignment of the event to a Partition should be handled.
+
+        type: object
+        $ref: '#/definitions/PartitionResolutionStrategy'
+
+      schema:
+        type: object
+        $ref: '#/definitions/EventTypeSchema'
+        description: |
+          The schema for this EventType. This is expected to be a json schema in yaml format (other formats
+          might be added in the future).
+
+      data_key_fields:
+        type: array
+        items:
+          type: string
+        description: |
+          Indicators of the path of the properties that constitute the primary key (identifier) of the data
+          within this Event.
+          If set MUST be a valid required field as defined in the schema.
+          (TBD should be required? Is applicable only to both Business and DataChange Events?)
+
+      ordering_key_field:
```

Comment:
@jkliff did we change this recently? 

Why only one field? Theoretically it can well be a composite key, the same way as `data_key_fields`. Also we have several times changed the name from `ordering` to `partitioning` and back. My last understanding was the we wanted to finalise on `partitioning_key` (and now seeing the composite key use-case: `partitioning_key_fields`).

Also later on we will have to formalise, what query language is used to extract the fields if they are not on the top level of the data.